samples/dnn/text_detection.cpp

```diff
@@ -72,23 +92,67 @@ int main(int argc, char** argv)
         }
 
         blobFromImage(frame, blob, 1.0, Size(inpWidth, inpHeight), Scalar(123.68, 116.78, 103.94), true, false);
-        net.setInput(blob);
-        net.forward(outs, outNames);
+        detector.setInput(blob);
+        tick_meter.start();
+        detector.forward(outs, outNames);
+        tick_meter.stop();
+
+        average_inference_time += tick_meter.getTimeMilli() / tick_meter.getCounter();
+        tick_meter.reset();
 
         Mat scores = outs[0];
         Mat geometry = outs[1];
 
         // Decode predicted bounding boxes.
         std::vector<RotatedRect> boxes;
         std::vector<float> confidences;
-        decode(scores, geometry, confThreshold, boxes, confidences);
+        decode_bounding_boxes(scores, geometry, confThreshold, boxes, confidences);
 
         // Apply non-maximum suppression procedure.
         std::vector<int> indices;
         NMSBoxes(boxes, confidences, confThreshold, nmsThreshold, indices);
 
-        // Render detections.
         Point2f ratio((float)frame.cols / inpWidth, (float)frame.rows / inpHeight);
+
+        // Render text.
+        for (size_t i = 0; i < indices.size() && !model_recognition.empty(); ++i)
+        {
+            RotatedRect& box = boxes[indices[i]];
+            Size2f box_size = box.size;
+
+            Point2f box_center = box.center;
+
+            box_size.width *= ratio.x;
+            box_size.height *= ratio.y;
+
+            box_center.x *= ratio.x;
+            box_center.y *= ratio.y;
+
+            Mat cropped;
+            four_points_transform(frame, box, box_size, ratio, cropped);
+
+            cvtColor(cropped, cropped, cv::COLOR_BGR2GRAY);
+
+            Mat blob_crop = blobFromImage(cropped, 1.0/127.5, Size(100,32), Scalar(127.5), false, false, CV_32F);
```

Comment:
Should be `Scalar::all(127.5)`. Ohterwise this is `[127.5, 0, 0, 0]`. Please also avoid specifying default arguments.